app/forms/batch_upsert_holidays.rb

```diff
@@ -0,0 +1,63 @@
+class BatchUpsertHolidays
+  include ActiveModel::Model
+  extend ActiveModel::Naming
+  include DateRangePickerHelper
+
+  attr_reader :title
+  attr_reader :date_range
+  attr_reader :users
+  attr_reader :all_day
+
+  validates :title, presence: true
+  validates :date_range, presence: true
+  validates :users, presence: true
+
+  def to_model
+    if @previous_holiday_ids.any?
+      @previous_holiday ||= Holiday.find(@previous_holiday_ids.first)
+    else
+      Holiday.new
+    end
+  end
+
+  def initialize(previous_holiday_ids = [], title = nil, all_day = true, date_range = nil, users = [])
```

Comment:
Can you move this under the `validates` section? Feels weird it being underneath another instance method.